packages/api/package.json

```diff
@@ -11,9 +11,12 @@
     "publish": "wrangler publish --env $(whoami)",
     "build": "webpack",
     "format": "prettier --write '**/*.{js,css,json,md}'",
-    "test": "say sorry"
+    "test": "playwright-test \"test/**/*.spec.js\" --sw src/index.js"
```

Comment:
How does this actually work? Is `playwright-test` actually injecting this as a service worker in a chrome? `src/index.js` isn't a valid service worker without some pre-processing... it's esm but it uses a node style import which I would expect to fail in a browser service worker context. Also how does it know what origin the service worker can catch fetch events to... and https and and.... so many questions.

Reply:
woosh. it runs it through esbuild!

https://github.com/hugomrdias/playwright-test/blob/096b773636311eff6038fe7a71fc0f38eea6221a/src/utils/build-sw.js#L20-L61